### PR TITLE
Add the validation of the last component in FilterByLastComponentDataset with tests

### PR DESCRIPTION
**Pull request by feihugis**
This PR adds the validation of the last component in `FilterByLastComponentDataset` and also adds the tests.

cc: @jsimsa

### CODE REVIEW

File: tensorflow/core/kernels/data/filter_by_component_dataset_op.cc [hunk 1]
```diff
@@ -130,6 +130,10 @@ class FilterByLastComponentDatasetOp : public UnaryDatasetOpKernel {
             return Status::OK();
           }
 
+          if (out_tensors->back().NumElements() != 1) {
```
Comment (feihugis) [inline tensorflow/core/kernels/data/filter_by_component_dataset_op.cc]: Do we need to check the data type of the last component as well?

Comment (jsimsa) [reply tensorflow/core/kernels/data/filter_by_component_dataset_op.cc]: yes, please check it is DT_BOOL

**Review by jsimsa — DISMISSED** [review]

**Comment by feihugis** [comment]
@jsimsa The dtype check for the last component is added in this commit (https://github.com/tensorflow/tensorflow/pull/28456/commits/8459281eebf1a8eccc9ea7904324fb991ca6cc55). It also removes the unnecessary test cases. Could you please have a look at the change?

**Review by jsimsa — DISMISSED** [review]
Change the message to "Last component must be a bool scalar.", otherwise LGTM.

**Comment by feihugis** [comment]
> Change the message to "Last component must be a bool scalar.", otherwise LGTM.

The message has been revised here (https://github.com/tensorflow/tensorflow/pull/28456/commits/c8d7dc556955d3d6939b79e67366bec88cde7844). Could you have a look?

**Review by jsimsa — APPROVED** [review]

**Pull request merged**